File: backend/Services/OrderService.cs
using AirmasterOrderApi.Models;

namespace AirmasterOrderApi.Services;

public class OrderService
{
    private readonly List<Product> _products = new()
    {
        new Product { ProductId = Guid.NewGuid(), Name = "Airmaster Running Shoes", Description = "High-performance shoes for urban runners.", Price = 129.99m, Category = "Footwear", StockQuantity = 200, ImageUrl = "https://example.com/images/shoes.png", UpdatedAt = DateTime.UtcNow },
        new Product { ProductId = Guid.NewGuid(), Name = "Airmaster Backpack", Description = "Durable travel backpack with smart compartments.", Price = 89.99m, Category = "Accessories", StockQuantity = 140, ImageUrl = "https://example.com/images/backpack.png", UpdatedAt = DateTime.UtcNow }
    };

    private readonly List<Order> _orders = new();

    public IEnumerable<Product> GetProducts() => _products;

    public Product? GetProduct(Guid id) => _products.FirstOrDefault(p => p.ProductId == id);

    public Order? GetOrder(Guid id) => _orders.FirstOrDefault(o => o.OrderId == id);

    public Order CreateOrder(OrderRequest request)
    {
        var order = new Order
        {
            OrderId = Guid.NewGuid(),
            UserId = request.UserId,
            CreatedAt = DateTime.UtcNow,
            Status = OrderStatus.Pending,
            ShippingAddress = request.ShippingAddress,
            BillingAddress = request.BillingAddress,
            Items = request.Items.ToList(),
            TotalAmount = request.Items.Sum(item => item.UnitPrice * item.Quantity)
        };

        _orders.Add(order);
        return order;
    }

    public PaymentResult ProcessPayment(PaymentRequest request)
    {
        var order = GetOrder(request.OrderId);
        if (order is null)
        {
            return new PaymentResult { Success = false, Message = "Order not found." };
        }

        // Simulated payment integration logic
        if (request.Amount <= 0)
        {
            return new PaymentResult { Success = false, Message = "Invalid payment amount." };
        }

        order.Status = OrderStatus.Paid;
        return new PaymentResult { Success = true, Message = "Payment successful.", TransactionId = Guid.NewGuid().ToString() };
    }
}

File: frontend/src/app/services/order.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Observable, BehaviorSubject } from 'rxjs';

export interface Product {
  productId: string;
  name: string;
  description: string;
  price: number;
  currency: string;
  imageUrl?: string;
}

export interface OrderItem {
  productId: string;
  quantity: number;
  unitPrice: number;
}

export interface OrderRequest {
  userId: string;
  items: OrderItem[];
  shippingAddress: string;
  billingAddress: string;
}

export interface Order {
  orderId: string;
  userId: string;
  createdAt: string;
  totalAmount: number;
  currency: string;
  status: string;
  shippingProvider: string;
  trackingNumber?: string;
  shippingAddress: string;
  billingAddress: string;
  items: OrderItem[];
}

export interface PaymentRequest {
  orderId: string;
  paymentProvider: string;
  token: string;
  amount: number;
  currency: string;
}

export interface PaymentResult {
  success: boolean;
  message: string;
  transactionId?: string;
}

export interface LoginRequest {
  email: string;
  password: string;
}

export interface AuthResponse {
  success: boolean;
  message: string;
  token: string;
  user: {
    userId: string;
    email: string;
    name: string;
    role: string;
  };
}

@Injectable({
  providedIn: 'root'
})
export class OrderService {
  private baseUrl = '/api';
  private authTokenSubject = new BehaviorSubject<string | null>(this.getToken());
  public authToken$ = this.authTokenSubject.asObservable();

  constructor(private http: HttpClient) {}

  private getToken(): string | null {
    return localStorage.getItem('authToken');
  }

  private setToken(token: string): void {
    localStorage.setItem('authToken', token);
    this.authTokenSubject.next(token);
  }

  private getHeaders(): HttpHeaders {
    const token = this.getToken();
    return new HttpHeaders({
      'Authorization': token ? `Bearer ${token}` : ''
    });
  }

  login(email: string, password: string): Observable<AuthResponse> {
    return this.http.post<AuthResponse>(`${this.baseUrl}/auth/login`, { email, password });
  }

  logout(): void {
    localStorage.removeItem('authToken');
    this.authTokenSubject.next(null);
  }

  isAuthenticated(): boolean {
    return !!this.getToken();
  }

  getProducts(): Observable<Product[]> {
    return this.http.get<Product[]>(`${this.baseUrl}/products`);
  }

  createOrder(request: OrderRequest): Observable<Order> {
    return this.http.post<Order>(`${this.baseUrl}/orders`, request, {
      headers: this.getHeaders()
    });
  }

  processPayment(request: PaymentRequest): Observable<PaymentResult> {
    return this.http.post<PaymentResult>(`${this.baseUrl}/payments`, request, {
      headers: this.getHeaders()
    });
  }

  getOrder(orderId: string): Observable<Order> {
    return this.http.get<Order>(`${this.baseUrl}/orders/${orderId}`, {
      headers: this.getHeaders()
    });
  }

  getUserOrders(userId: string) {
    return this.http.get<Order[]>(`${this.baseUrl}/orders?userId=${userId}`, {
      headers: this.getHeaders()
    });
  }

  shipOrder(orderId: string) {
    return this.http.post<Order>(`${this.baseUrl}/orders/${orderId}/ship`, {}, {
      headers: this.getHeaders()
    });
  }

  getAnalytics() {
    return this.http.get(`${this.baseUrl}/admin/analytics`, {
      headers: this.getHeaders()
    });
  }
}
